das/checkpoint.go

```diff
@@ -48,3 +48,17 @@ func (c checkpoint) String() string {
 
 	return str
 }
+
+// WorkersTotalSampled returns the total amount of sampled headers
+func (c checkpoint) WorkersTotalSampled() int {
+	var total int
+	for _, w := range c.Workers {
+		total += int(w.To - w.From)
+	}
+	return total
+}
```

Comment:
Name of the function doesn't match the logic. You calculating amount of not sampled items, that workers will pick up upon resume. Those headers are not yet sampled.

Also if w.To == w.From there is 1 header to be sampled. So total should be like this:
```
total += int(w.To - w.From) + 1
```

Reply:
I'm slightly confused, what you are saying regarding the "to be sampled" ranges for workers, makes sense, however in your [previous comment](https://github.com/celestiaorg/celestia-node/pull/1638#discussion_r1087757155) you mentioned that `w.To - w.From` would be the all done from workers.

So which is correct, `c.SampleFrom - 1 - allFailed` or should add to these the workers' to and from? I'm thinking the first formula.

Reply:
The comment you mentioned doesn't say w.To - w.From would be the all done. It is items that were not finished, and will be continued after restart by resumed workers.  Thus sum of those needs to be subtracted from current allSampled value.

Reply:
Thank you for the elaboration @walldis, as you've helped point out through discussion, the correct formula is:

```
totalSampled = c.SampleFrom - TotalInProgress - TotalFailed
```